numpy-stubs/__init__.pyi

```diff
@@ -813,14 +814,27 @@ _ScalarNumpy = Union[generic, dt.datetime, dt.timedelta]
 _ScalarBuiltin = Union[str, bytes, dt.date, dt.timedelta, bool, int, float, complex]
 _Scalar = Union[_ScalarBuiltin, _ScalarNumpy]
 
-_ScalarGeneric = TypeVar(
-    "_ScalarGeneric", bound=Union[dt.datetime, dt.timedelta, generic]
+# Integers and booleans can generally be used interchangeably
+_ScalarInt = TypeVar("_ScalarInt", bound=Union[integer, bool_])
```

Comment:
I don't think `_ScalarInt` is the best name here; seems like it should be `_ScalarIntOrBool`. I mostly don't want this to accidentally get cargo-culted to other places where NumPy int/bool diverge; e.g. arithmetic.

Reply:
Based on https://github.com/numpy/numpy-stubs/pull/74#discussion_r416159856 I'd argue that it is a suitable name. It can still be changed though if you don't agree,

Besides, for arithmetic it would (most likely) be more convenient to use a typevar of ``np.number`` anyway.